### PR TITLE
test(conformance): all tests enabled

### PR DESCRIPTION
**Pull request by mlavacca**
**What this PR does / why we need it**:

The old way of running conformance tests has been replaced by `csuite.Run`. Somehow, with the old method we did not run all the tests, hence we couldn't see the eventual failures. By using the new method, it popped up that we need to fix 
- `HTTPRouteMatching`,
- `HTTPRouteRedirectHostAndStatus`,

in addition to the already known 
- `HTTPRouteHeaderMatching`,

Additionally, all the optional features have been enabled and the additional failing tests have been disabled to clearly state that we need to fix them sooner or later.

**Which issue this PR fixes**:

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: test/conformance/gateway_conformance_test.go [hunk 1]
```diff
@@ -25,8 +24,9 @@ import (
 )
 
 var (
-	showDebug     = true
-	shouldCleanup = true
+	showDebug                  = true
+	shouldCleanup              = true
+	enableAllSupportedFeatures = true
```
Comment (czeslavo) [inline test/conformance/gateway_conformance_test.go:29]: Nit: could we make these variables consts?

Comment (mlavacca) [reply test/conformance/gateway_conformance_test.go:29]: 👍

**Review by czeslavo — APPROVED** [review]
Just to make sure: do we have any tracking issues for the extended and experimental conformance levels?

**Comment by mlavacca** [comment]
> Just to make sure: do we have any tracking issues for the extended and experimental conformance levels?

I added the links to the GH issues in the code as comments

**Review by czeslavo — APPROVED** [review]

**Review by pmalek — APPROVED** [review]

**Pull request merged**